### PR TITLE
VPN-6356: Fix assert in QTest::mouseEvent()

### PR DESCRIPTION
**Pull request by oskirby**
## Description
I have been seeing a bunch of auth functional tests failing in CI due to an assert in `QTest::mouseEvent()` due to a null `QWindow` pointer. After some digging, I think this is really a consequence of the bug noted by [VPN-3697](https://mozilla-hub.atlassian.net/browse/VPN-3697), where we encounter a race condition between the rendering thread and the GUI thread.

To resolve this, we should poke the rendering thread and wait for the frame draw to finish before submitting the click to the window. This should ensure that the QML scene graph is synchronized with the QML content before we process the click event.

Some subtle changes that revealed some bugs:
- Sending the click to `item->window()` can actually click things that are offscreen, and sending the click to the root window revealed a handful of test bugs.
- There were a lot of test bugs that were being hidden by the `QTest::qWait()` workaround.

## Reference
JIRA issue: [VPN-3697](https://mozilla-hub.atlassian.net/browse/VPN-3697)
JIRA issue: [VPN-6356](https://mozilla-hub.atlassian.net/browse/VPN-6356)

## Checklist
    
- [ ] My code follows the style guidelines for this project
- [ ] I have not added any packages that contain high risk or unknown licenses (GPL,  LGPL, MPL, etc. consult with DevOps if in question)
- [ ] I have performed a self review of my own code
- [ ] I have commented my code PARTICULARLY in hard to understand areas
- [ ] I have added thorough tests where needed

[VPN-3697]: https://mozilla-hub.atlassian.net/browse/VPN-3697?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[VPN-3697]: https://mozilla-hub.atlassian.net/browse/VPN-3697?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[VPN-6356]: https://mozilla-hub.atlassian.net/browse/VPN-6356?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/inspector/inspectorhandler.cpp [hunk 1]
```diff
@@ -357,6 +358,15 @@ static QList<InspectorCommand> s_commands{
                      [](InspectorHandler*, const QList<QByteArray>& arguments) {
                        QJsonObject obj;
 
+                       // Ensure window rendering is finished.
+                       QQuickWindow* window = qobject_cast<QQuickWindow*>(
+                          QmlEngineHolder::instance()->window());
+                       QSignalSpy spy(window, SIGNAL(afterRendering()));
+                       window->update();
+                       if (!spy.wait(150)) {
+                        logger.debug() << "Never got an afterRendering signal";
```
Comment (brizental) [inline src/inspector/inspectorhandler.cpp]: nit: logger.warn

File: src/inspector/inspectorhandler.cpp [hunk 2]
```diff
@@ -357,6 +358,19 @@ static QList<InspectorCommand> s_commands{
                      [](InspectorHandler*, const QList<QByteArray>& arguments) {
                        QJsonObject obj;
 
+                       QQuickWindow* window = qobject_cast<QQuickWindow*>(
+                           QmlEngineHolder::instance()->window());
+#ifndef MZ_WASM
```
Comment (brizental) [inline src/inspector/inspectorhandler.cpp:363]: nit: Please add a comment here on why WASM is being skipped.

File: src/ui/screens/home/servers/ServerCountry.qml [hunk 3]
```diff
@@ -22,7 +22,7 @@ MZClickableRow {
     property var currentCityIndex
     property alias serverCountryName: countryName.text
     property var cityList: cityListVisible ? cityLoader.item : cityLoader
-    property var busy: cityListVisible ? scrollAnimation.running : false
+    property var busy: cityListVisible && (scrollAnimation.running || listOpenTransition.running || listClosedTransition.running)
```
Comment (brizental) [inline src/ui/screens/home/servers/ServerCountry.qml:25]: nit: Does something like this work on QML?

```suggestion
    property var busy: cityListVisible && transitions.find(t => t.running)
```

**Review by brizental — APPROVED** [review]

File: src/ui/screens/home/servers/ServerCountry.qml (hunk 3 above)
Comment (oskirby) [reply src/ui/screens/home/servers/ServerCountry.qml:25]: This construction does not seem to work in QML. It generates an error:
`qrc:/ui/screens/home/servers/ServerCountry.qml:25: TypeError: Property 'find' of object [object Object] is not a function (ServerCountry.qml:25)`

File: src/inspector/inspectorhandler.cpp (hunk 2 above)
Comment (oskirby) [reply src/inspector/inspectorhandler.cpp]: The short answer is that this results in a crash on WASM, and this workaround only seems to be needed on systems that use multithreaded rendering engines (of which WASM is not). I will add some more comments and make the intent a little clearer.

**Review by strseb — APPROVED** [review]

File: src/ui/screens/home/servers/ServerCountry.qml (hunk 3 above)
Comment (brizental) [reply src/ui/screens/home/servers/ServerCountry.qml:25]: Looks like translations is not an array although it looks like one. 🤷‍♀️

**Pull request merged**